### DISCUSSION TITLE
How to write the registry

### DISCUSSION THREAD

**Discussion by MayueCif** [dotnet/maui#26209] [Q&A]
### Description

Is it not allowed to manipulate the registry with maui?

### Steps to Reproduce

`
RegistryKey registry = xxx();
registry.SetValue("Key", 1,RegistryValueKind.DWord);
//registry.Flush(); 
registry.Close();
`

**Comment by MayueCif** [comment]
any suggestions？

**Comment by FaithfulDev** [comment]
It works fine for me. Did you get an exception or something?

This is my snippet:
```cs
RegistryKey registry = Registry.CurrentUser.CreateSubKey(@"Software\MySoftware");
registry.SetValue("Key", 1, RegistryValueKind.DWord);
registry.Flush();
registry.Close();
```

**Reply by MayueCif** [reply]
a value was written.but value is dif from registry.exe view.understand

![Snipaste_2024-12-01_23-12-35](https://github.com/user-attachments/assets/9bca18be-47f2-4943-9277-a36472c48035)
![Snipaste_2024-12-01_23-11-40](https://github.com/user-attachments/assets/da79b2af-89fd-4b1e-b06e-534f31ac57bd)
![Snipaste_2024-12-01_23-11-57](https://github.com/user-attachments/assets/30534ad1-5875-4563-bb9d-c11b27504883)

**Reply by FaithfulDev** [reply]
Here is what I tried:
I wrote "1" to the registry. Checked the value with regedit.exe (correct value). Then I updated the value to "2" and checked again with the regedit.exe (again, correct value). `GetValue` always returned the correct value.

Remember to refresh the regedit.exe view, it does not auto-refresh (at least my one didn't).

Sorry, but I can't reproduce your issue 😞 

Idea: Is there maybe a different program updating the key as well in the background? Did you try it out with a completely different key?

**Comment by MayueCif** [comment]
I have found an issue, I cannot find the name I wrote. The old name was ProxyEnable, but I have changed it to ProxyEmpower_1. I cannot find this name in regedit.exe search anymore 🤨
